agent/agent.go

```diff
@@ -112,6 +112,10 @@ LOOP:
 		case <-timer.C:
 			nextTask, err := a.comm.GetNextTask(ctx, &apimodels.GetNextTaskDetails{tc.taskGroup})
 			if err != nil {
+				// task secret doesn't match, get another task
+				if err.Error() == client.HTTPConflictError {
```

Comment:
don't call Error() but use errors.Cause(err) and actually have the method return HTTPConflictError which should be an error type.

Reply:
Done (I think)